tests/e2e/utils/network_utils.go

```diff
@@ -21,11 +21,19 @@ import (
 	"fmt"
 	"strconv"
 
+	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2017-09-01/network"
 	aznetwork "github.com/Azure/azure-sdk-for-go/services/network/mgmt/2017-09-01/network"
 	"github.com/Azure/go-autorest/autorest/to"
 	"k8s.io/apimachinery/pkg/util/wait"
 )
 
+type PublicIPAddressesClient interface {
```

Comment:
Why do we need these two? there's already `AzureTestClient`

Reply:
Please use `azureTestClient.createPublicIPAddressesClient()` to create the PublicIPAddressesClient